From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Thu, 24 Sep 2026 10:57:53 -0400
Subject: [PATCH] cmake: Use same naming rules for iio-compat as v0 libiio

This adds a check for APPLE targets and sets the iio-compat library
suffix to .0.dylib to match the output from the v0 libiio build.

Signed-off-by: Ryan Volz <ryan.volz@gmail.com>
---
 CMakeLists.txt | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 105c4ea8..91d8cd9f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -165,7 +165,9 @@ if (CMAKE_C_COMPILER_ID MATCHES "GNU|Clang|MSVC")
                        NO_SONAME ON
                        VERSION 0.99
                        SOVERSION 0)
-        if (NOT WIN32)
+        if (APPLE)
+            set_target_properties(iio-compat PROPERTIES SUFFIX .0.dylib)
+        elseif (NOT WIN32)
             set_target_properties(iio-compat PROPERTIES SUFFIX .so.0)
         endif ()
         if (WITH_GCOV)
-- 
2.51.0

